src/melody/search.py

```diff
@@ -39,11 +39,48 @@
 class SearchMethod(object):
     '''A utility baseclass for different search/optimisation methods'''
 
-    def __init__(self, function=None, inputs=None, state=None):
-        self._function = function
+    def __init__(self, inputs, function, state=None):
         self._inputs = inputs
+        self._check_inputs()
+        self._function = function
+        self._check_function()
         self._state = state
 
+    def _check_inputs(self):
+        ''' make some basic checks on the inputs to make sure they are valid'''
+        try:
+            _ = self._inputs[0]
+        except TypeError:
+            raise RuntimeError(
+                "inputs should be iterable but found type='{0}', value="
+                "'{1}'".format(type(self._inputs), str(self._inputs)))
+        from melody.inputs import Input
+        for check_input in self._inputs:
+            if not isinstance(check_input, Input):
+                raise RuntimeError(
+                    "input should be a subclass of the Input class but "
+                    "found type='{0}', value='{1}'".format(type(check_input),
+                                                           str(check_input)))
+
+    def _check_function(self):
+        ''' make some basic checks on the function to make sure it is valid'''
+        # note, callable is valid for Python 2 and Python 3.2 onwards but
```

Comment:
Is python 2 referring to 2.7 or more generally all versions of python 2 users might have? 2.6 seems to be still cropping up in the wild